sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DoNotOverwriteCollectionElements.vb

```diff
@@ -4,214 +4,232 @@ Imports System.Linq
+            dict.Add(0, 1) ' Noncompliant
+        End Sub
+
+        Private Sub IDictionaryAddOnMultiMemberAccess(ByVal c As TestCases)
+            ' The below code will throw an ArgumentException at runtime
+            c.dictionaryField.Add(0, 0) ' Secondary
+            c.dictionaryField.Add(0, 1) ' Noncompliant
+        End Sub
+
+        Private Sub DoNotReportOnNonDictionaryAdd(ByVal c As CustomAddItem)
+            c.Add(0, 1)
+            c.Add(0, 2) ' Compliant this is not on a dictionary
+        End Sub
+
+        Private Sub MeWithDifferent()
+            Me.dictionaryField(0) = 1 ' Compliant
+            Me.dictionaryField2(0) = 1 ' Compliant
+        End Sub
+
+        Private Sub MyClassWithDifferent()
+            MyClass.dictionaryField(0) = 1 ' Compliant
+            MyClass.dictionaryField2(0) = 1 ' Compliant
+        End Sub
+
+        Private Sub ArrayFields()
+            fooList(0).FooField = 1
+            fooList(0).BarField = 1
+            AddHandler fooList(0).Ev1, AddressOf Handler
+            AddHandler fooList(0).Ev2, AddressOf Handler
+        End Sub
+
+        Private Sub Handler()
+        End Sub
+
+        Private Sub ConditionalAccess1()
+            dictionaryField.Add(0, 1) ' Secondary
+            dictionaryField?.Add(0, 1) ' Noncompliant
+        End Sub
+
+        Private Sub ConditionalAccess2()
+            dictionaryField?.Add(0, 1) ' Secondary
+            dictionaryField.Add(0, 1) ' Noncompliant
+        End Sub
+
+    End Class
+
+    Class InheritTestCases
+        Inherits TestCases
+
+        Private Sub BaseSame()
+            MyBase.dictionaryField(0) = 1 ' Secondary
+            MyBase.dictionaryField(0) = 1 ' Noncompliant
+        End Sub
+
+        Private Sub BaseDifferent()
+            MyBase.dictionaryField(0) = 1 ' Compliant
+            MyBase.dictionaryField2(0) = 1 ' Compliant
+        End Sub
+
+    End Class
+
+    Class Foo
+
+        Public FooField
+        Public BarField
+        Public Event Ev1()
+        Public Event Ev2()
+
+    End Class
+
+    Class CustomIndexerOneArg
+
+        Default Property Item(ByVal key As String) As Integer
+            Get
+                Return 1
+            End Get
+            Set
+
+            End Set
+        End Property
+
+    End Class
+
+    Class CustomIndexerMultiArg
+
+        Default Property Item(ByVal s As String, ByVal i As Integer, ByVal d As Double) As Integer
+            Get
+                Return 1
+            End Get
+            Set
+
+            End Set
+        End Property
+
+    End Class
+
+    Class CustomAddItem
+
+        Public Sub Add(ByVal a As Integer, ByVal b As Integer)
+        End Sub
+
+    End Class
 
     'See https://github.com/SonarSource/sonar-dotnet/issues/2235
     Class NullReferenceReproducer
+
         Sub FooBar()
-            Bar ' AD0001 NullReferenceException in GetFirstArgumentExpression
+            Bar() ' AD0001 NullReferenceException in GetFirstArgumentExpression
```

Comment:
Formatting also removed the purpose of the test here. The method call without parenthesis was added to make sure that rule does not throw `NullReferenceException` when method invocation is done without argument. See PR #2237

Reply:
Same as above